pkg/hs/urbit-king/lib/Urbit/Vere/Dawn.hs

```diff
@@ -168,122 +153,114 @@ dawnPostRequests endpoint responseBuilder requests = do
   toFullRequest (rid, req) = object [ "jsonrpc" .= ("2.0" :: Text)
                                     , "method"  .= getRequestMethod req
                                     , "params"  .= req
-                                    , "id"      .= rid
+                                    , "id"      .= (show rid)
                                     ]
 
 -- Azimuth JSON Requests -------------------------------------------------------
 
--- Not a full implementation of the Ethereum ABI, but just the ability to call
--- a method by encoded id (like 0x63fa9a87 for `points(uint32)`), and a single
--- UIntN 32 parameter.
-encodeCall :: Text -> Int -> Text
-encodeCall method idx = method <> leadingZeroes <> renderedNumber
-  where
-    renderedNumber = pack $ showHex idx ""
-    leadingZeroes = replicate (64 - length renderedNumber) '0'
-
-data BlockRequest = BlockRequest
-  deriving (Show, Eq)
-
-instance RequestMethod BlockRequest where
-  getRequestMethod BlockRequest = "eth_blockNumber"
-
-instance ToJSON BlockRequest where
-  toJSON BlockRequest = Array $ fromList []
-
--- No need to parse, it's already in the format we'll pass as an argument to
--- eth calls which take a block number.
-parseBlockRequest :: BlockRequest -> Text -> TextBlockNum
-parseBlockRequest _ txt = txt
-
-type TextBlockNum = Text
-
-data PointRequest = PointRequest
-  { grqHexBlockNum :: TextBlockNum
-  , grqPointId :: Int
-  } deriving (Show, Eq)
+data PointResponse = PointResponse
+  --NOTE  also contains dominion and ownership, but not actually used here
+  { prNetwork :: PointNetwork
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointResponse where
+  parseJSON (Object o) = do
+    prNetwork <- o .: "network"
+    pure PointResponse{..}
+  parseJSON _ = do error "failed to parse PointResponse json"
+
+data PointNetwork = PointNetwork
+  { pnKeys :: PointKeys
+  , pnSponsor :: PointSponsor
+  , pnRift :: ContNum
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointNetwork where
+  parseJSON (Object o) = do
+    pnKeys <- o .: "keys"
+    pnSponsor <- o .: "sponsor"
+    pnRift <- o .: "rift"
+    pure PointNetwork{..}
+  parseJSON _ = do error "failed to parse PointNetwork json"
+
+data PointKeys = PointKeys
+  { pkLife :: Life
+  , pkSuite :: Int
+  , pkAuth :: Text  --NOTE  0xhex string
+  , pkCrypt :: Text  --NOTE  0xhex string
```

Comment:
Okay so these should be numbers (either Atom or Natural), and you should parse the hex strings as part of the FromJSON rather than in the later step. Because if they're not hex string, that's a parse error and should be handled in the parse step, rather than crashing the king in `passFromText`

Reply:
Also please do remember to put a space after `--` and a single space after NOTE